cli/flox/src/commands/environment.rs

```diff
@@ -570,25 +612,171 @@ impl Activate {
             },
         };
 
-        if !self.run_args.is_empty() {
+        if !run_args.is_empty() {
             command.arg("-i");
             command.arg("-c");
-            command.arg(self.run_args.join(" "));
+            command.arg(run_args.join(" "));
         }
 
         debug!("running activation command: {:?}", command);
 
-        if self.run_args.is_empty() {
+        if run_args.is_empty() {
             let message = formatdoc! {"
                 ✅  You are now using the environment {now_active}.
                 To stop using this environment, type 'exit'"};
             info!("{message}");
         }
-        let error = command.exec();
-
         // exec should never return
+        command.exec().into()
+    }
+
+    /// Patch the PATH to undo the effects of `/usr/libexec/path_helper`
+    ///
+    /// Darwin has a "path_helper" which indiscriminately reorders the path
+    /// to put the system curated path items first in the `PATH`,
+    /// which completely breaks the user's ability to manage their `PATH` in subshells,
+    /// e.g. when using tmux.
+    ///
+    /// On macos `/usr/libexec/path_helper` is typically invoked from
+    /// default shell rc files, e.g. `/etc/profile` and `/etc/zprofile`.
+    ///
+    /// This function attempts to undo the effects of `/usr/libexec/path_helper`
+    /// by partitioning the `PATH` into two parts:
+    /// 1. known paths of activated flox environments
+    ///    and nix store paths (e.g. `/nix/store/...`) -- assumed to be `nix develop` paths
+    /// 2. everything else
+    ///
+    /// The `PATH` is then reordered to put the flox environment and nix store paths first.
+    /// The order within the two partitions is preserved.
+    fn fixup_path(flox_env_dirs: IndexSet<PathBuf>) -> Option<Vec<PathBuf>> {
+        if !Path::new("/usr/libexec/path_helper").exists() {
+            return None;
+        }
+        let path_var = env::var("PATH").unwrap_or_default();
+        Some(Self::fixup_path_with(path_var, flox_env_dirs))
+    }
+
+    /// Patch a given PATH value to undo the effects of `/usr/libexec/path_helper`
+    ///
+    /// See [Self::fixup_path] for more details.
+    fn fixup_path_with(
+        path_var: impl AsRef<OsStr>,
+        flox_env_dirs: IndexSet<PathBuf>,
+    ) -> Vec<PathBuf> {
+        let path_iter = env::split_paths(&path_var);
+
+        let (flox_and_nix_path, path) = path_iter.partition::<Vec<_>, _>(|path| {
+            let is_flox_path = path
+                .parent()
+                .map(|path| flox_env_dirs.contains(path))
+                .unwrap_or_else(|| flox_env_dirs.contains(path));
+
+            path.starts_with("/nix/store") || is_flox_path
```

Comment:
nonblocking: doesn't seem like we need to re-order `/nix/store` paths anymore because we don't use `nix develop`

Reply:
I think since flox still ships with nix it's fine for now to keep this in.

Reply:
If I'm a `nix` user who also uses `flox` ( mostly of interest for `flox` staff but not exclusively ); will using `/nix/store/` prefixes to detect paths we need to patch cause issues?

I suspect it will based on the reading these inline comments.

Reply:
it should not, the order of `{flox env paths + nix store paths}` and `{ other paths }` is preserved.

```<path A>:/nix/store/.../:/my/env/.flox/run/bin:<path B>```

will be reordered as 

```
/nix/store/.../:/my/env/.flox/run/bin:<path A>:<path B>
```

this messes with your own custom path settings (e.g. `path A`) but so does `path_helper`.